doc/specs/Keyboard-Selection.md

```diff
@@ -0,0 +1,141 @@
+
+For `SelectionExpansionMode = Viewport`, the selection anchor will be updated according to the viewport's height. Horizontal movements will be updated according to the viewport's width, thus resulting in the anchor being moved to the left/right boundary of the viewport.
+
+**NOTE**: An important thing to handle properly in all cases is wide glyphs. The user should not be allowed to select a portion of a wide glyph; it should be all or none of it. When calling `_ExpandWideGlyphSelection` functions, the result must be saved to the anchor.
+
+**NOTE**: In all cases, horizontal movements attempting to move past the left/right viewport boundaries result in a wrap. Vertical movements attempting to move past the top/bottom viewport boundaries will scroll such that the selection is at the edge of the screen. Vertical movements attempting to move past the top/bottom buffer boundaries will be clamped to be within buffer boundaries.
+
+Every combination of the `Direction` and `SelectionExpansionMode` will map to a keybinding. These pairings are shown below in the UI/UX Design --> Keybindings section.
+
+### Mark Mode
+Upon mark mode being enabled by a user keybinding, the current cursor position becomes the selection anchor. The arrow keys move the selection anchor by cell.
+
+
+## UI/UX Design
+
+### Keybindings
+
+By default, the following keybindings will be set:
+| Action | Keychord | `KeyboardSelection(Direction, SelectionExpansionMode)` Parameters |
+|--|--|--|
+| `MoveSelectionAnchorUp`              | `Shift` + `↑`              | `Up`      , `Cell`
+| `MoveSelectionAnchorDown`            | `Shift` + `↓`              | `Down`    , `Cell`
+| `MoveSelectionAnchorLeft`            | `Shift` + `←`              | `Left`    , `Cell`
+| `MoveSelectionAnchorRight`           | `Shift` + `→`              | `Right`   , `Cell`
+| `MoveSelectionAnchorUpOneScreen`     | `Shift` + `Page Up`        | `Up`      , `Viewport`
+| `MoveSelectionAnchorDownOneScreen`   | `Shift` + `Page Down`      | `Down`    , `Viewport`
+| `MoveSelectionAnchorToLeftMargin`    | `Shift` + `Home`           | `Left`    , `Viewport`
+| `MoveSelectionAnchorToRightMargin`   | `Shift` + `End`            | `Right`   , `Viewport`
+| `MoveSelectionAnchorUpByWord`        | `Ctrl` + `Shift` + `↑`     | `Up`      , `Word`
+| `MoveSelectionAnchorDownByWord`      | `Ctrl` + `Shift` + `↓`     | `Down`    , `Word`
+| `MoveSelectionAnchorLeftByWord`      | `Ctrl` + `Shift` + `←`     | `Left`    , `Word`
+| `MoveSelectionAnchorRightByWord`     | `Ctrl` + `Shift` + `→`     | `Right`   , `Word`
+| `MoveSelectionAnchorToBufferStart`   | `Ctrl` + `Shift` + `Home`  | N/A

```

Comment:
Should there be a fourth `SelectionExpansionMode`, `Buffer` perhaps, for these three (`MoveSelectionAnchorToBufferStart`, `MoveSelectionAnchorToBufferEnd`, `SelectEntireBuffer`)? (this is a thought question, not a _change this_ comment)

Reply:
That might be overkill since `MoveSelectionAnchorToBufferStart` just sets the anchor to `{0,0}`, and `MoveSelectionAnchorToBufferEnd` sets it to `textBuffer.dimensions()` (then `SelectEntireBuffer` doing both).